### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by scroll17**

### CODE REVIEW

**Review by olexandr-mazepa — CHANGES_REQUESTED** [review]

File: mailjet-client.js [hunk 1]
```diff
@@ -224,12 +212,12 @@ MailjetClient.prototype.connectStrategy = function (apiKey, apiSecret, options)
 }
 
 MailjetClient.prototype.setConfig = function (options) {
-  const config = require('./config.json')
-  if (typeof options === 'object' && options != null && options.length != 0) {
+  const config = { ...require('./config.json') }
+  if (typeof options === 'object' && options !== null) {
```
[the remark this answers is not in the document]
Comment (scroll17) [reply mailjet-client.js:216]: It's make sense, but i think that not need at the moment

File: mailjet-client.js [hunk 2]
```diff
@@ -224,12 +212,12 @@ MailjetClient.prototype.connectStrategy = function (apiKey, apiSecret, options)
...
     if (options.url) config.url = options.url
     if (options.version) config.version = options.version
-    if (options.perform_api_call) config.perform_api_call = options.perform_api_call
-  } else if (options != null) {
+    if ('perform_api_call' in options) config.perform_api_call = options.perform_api_call
+  } else if (options != null) { // TODO: Bug -> must be strict equal
```
[the remark this answers is not in the document]
Comment (scroll17) [reply mailjet-client.js:220]: I am planning to fix it in next iteration

File: mailjet-client.js [hunk 3]
```diff
@@ -193,25 +179,27 @@ MailjetClient.prototype.connectStrategy = function (apiKey, apiSecret, options)
 
   function basicConnectStrategy(apiKey, apiSecret, options) {
     if (!apiKey) {
-      throw new Error('Mailjet API_KEY is required');
+      throw new Error('Mailjet API_KEY is required')
```
[the remark this answers is not in the document]
Comment (scroll17) [reply mailjet-client.js:182]: Yeah, I agree with you, semicolons it's good practice

File: package.json [hunk 4]
```diff
@@ -6,6 +6,12 @@
   "directories": {
     "test": "test"
   },
+  "scripts": {
+    "test:unit": "mocha test/unit/*",
+    "test:int": "mocha test/integration/*",
+    "test": "mocha --recursive",
```
[the remark this answers is not in the document]
Comment (scroll17) [reply package.json:10]: Yes, it's was be added

**Review by olexandr-mazepa — APPROVED** [review]

**Pull request merged**